src/main/java/com/google/devtools/build/lib/analysis/config/ExecutionInfoModifier.java

```diff
@@ -103,6 +107,28 @@ public boolean matches(String mnemonic) {
     return expressions().stream().anyMatch(expr -> expr.pattern().matcher(mnemonic).matches());
   }
 
+  /**
+   * Merge all the elements of {@code executionInfoList} into a single {@link ExecutionInfoModifier}.
+   * The expressions in the returned instance may contain the same pattern multiple times.
+   */
+  public static ExecutionInfoModifier collapse(List<ExecutionInfoModifier> executionInfoList, boolean isAdditive) {
+    ImmutableList.Builder<Expression> allExpressionsBuilder = ImmutableList.builder();
+
+    if (executionInfoList != null && executionInfoList.size() > 0) {
+      if (isAdditive) {
+        for (ExecutionInfoModifier eim : executionInfoList) {
+          allExpressionsBuilder.addAll(eim.expressions());
+        }
+      } else {
+        // When not treating execution info as additive, only the last passed option wins.
+        allExpressionsBuilder.addAll(executionInfoList.get(executionInfoList.size() - 1).expressions());
+      }
+
+    }
+
+    return create("", allExpressionsBuilder.build());
```

Comment:
We want to pass the original input from the command line to the first argument because it is used for equality check. Maybe join `eim.option()` with `,`? Please also assert the value of `option()` for the merged modifier in the test cases.

Reply:
Thanks - updated 👍